### ISSUE TITLE
Deprecation warnings due to usage of Fixnum, Bignum constants

### ISSUE THREAD

**Issue by aergonaut** [cognitect/transit-ruby#22]
With Ruby 2.4.1 and transit-ruby 0.8.599, I get deprecation warnings from Transit about references to Fixnum and Bignum, which are [deprecated in Ruby 2.4.1+](https://bugs.ruby-lang.org/issues/12005).

```
% ruby --version
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin15]
```

```
/Users/chrisfung/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/transit-ruby-0.8.599/lib/transit/write_handlers.rb:418: warning: constant ::Fixnum is deprecated
/Users/chrisfung/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/transit-ruby-0.8.599/lib/transit/write_handlers.rb:419: warning: constant ::Bignum is deprecated
```

Is there a plan to update that file to remove the references to Fixnum and Bignum and replace them with just Integer?

**Comment by dchelimsky** [comment]
I'll try to take a look at this and other open issues Friday.

**Comment by dchelimsky** [comment]
I've got this addressed on a branch but I have to do some more testing across Ruby versions (specifically JRuby versions) before releasing. Probably won't get to that before this coming Friday, but it's in progress.

**Comment by dchelimsky** [comment]
Fixed in e8caa8d
Released in transit-ruby-0.8.602
